// @dasha.ai/sdk/dist/rest-api/group.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GroupApi = void 0;
const errors = __importStar(require("../errors"));
const log_1 = require("../log");
const base = __importStar(require("./base"));
const api = __importStar(require("./generated/core/api"));
class GroupApi extends base.Api {
    constructor(account, log) {
        super(api.GroupApi, account, log);
    }
    async createGroup(request) {
        const response = await this.api.apiV1GroupPost(request, {
            validateStatus: (status) => status < 400 || status === 409,
        });
        if (response.status === 409) {
            log_1.logAndThrow(this.log, new errors.GroupExistsError(request.name));
        }
        return response.data;
    }
    async createTemporaryGroup() {
        const response = await this.api.apiV1GroupTemporaryPost({}, {
            validateStatus: (status) => status < 400,
        });
        return response.data;
    }
    async deleteGroup(name, options) {
        const getResponse = await this.api.apiV1GroupByNameGet(name, {
            validateStatus: (status) => status < 400 || status === 404,
        });
        if (getResponse.status === 404) {
            if (!(options === null || options === void 0 ? void 0 : options.quiet))
                log_1.logAndThrow(this.log, new errors.GroupNotFoundError(name));
            return;
        }
        await this.api.apiV1GroupIdDelete(getResponse.data.id, {
            validateStatus: (status) => status < 400,
        });
    }
    async getDefaultGroupName() {
        const defaultGroupIdResponse = await this.api.apiV1GroupDefaultGet({
            validateStatus: (status) => status < 400,
        });
        const groupResponse = await this.api.apiV1GroupIdGet(defaultGroupIdResponse.data, {
            validateStatus: (status) => status < 400,
        });
        return groupResponse.data.name;
    }
    /** should only be called with a valid group id */
    async getGroupById(id) {
        const response = await this.api.apiV1GroupIdGet(id, {
            // we are content to just throw an InternalError on 404
            validateStatus: (status) => status < 400,
        });
        return response.data;
    }
    async getGroupByName(name) {
        const response = await this.api.apiV1GroupByNameGet(name, {
            validateStatus: (status) => status < 400 || status === 404,
        });
        if (response.status === 404) {
            log_1.logAndThrow(this.log, new errors.GroupNotFoundError(name));
        }
        return response.data;
    }
    async listGroups(options) {
        const response = await this.api.apiV1GroupGet(undefined, // skip
        undefined, // take
        options === null || options === void 0 ? void 0 : options.includeTemporary, {
            validateStatus: (status) => status < 400,
        });
        return response.data;
    }
    async updateGroup(name, request) {
        const getResponse = await this.api.apiV1GroupByNameGet(name, {
            validateStatus: (status) => status < 400 || status === 404,
        });
        if (getResponse.status === 404) {
            log_1.logAndThrow(this.log, new errors.GroupNotFoundError(name));
        }
        const patchResponse = await this.api.apiV1GroupIdPatch(getResponse.data.id, request, {
            validateStatus: (status) => status < 400 || status === 409,
        });
        if (patchResponse.status === 409) {
            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
            log_1.logAndThrow(this.log, new errors.GroupExistsError(request.name));
        }
        return patchResponse.data;
    }
}
exports.GroupApi = GroupApi;
