// @dasha.ai/sdk/dist/rest-api/generated/core/api/misc-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Dasha.AI Platform API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MiscApi = exports.MiscApiFactory = exports.MiscApiFp = exports.MiscApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * MiscApi - axios parameter creator
 * @export
 */
const MiscApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1MiscAuthServerGet: async (options = {}) => {
            const localVarPath = `/api/v1/misc/authServer`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            await common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            await common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.MiscApiAxiosParamCreator = MiscApiAxiosParamCreator;
/**
 * MiscApi - functional programming interface
 * @export
 */
const MiscApiFp = function (configuration) {
    const localVarAxiosParamCreator = exports.MiscApiAxiosParamCreator(configuration);
    return {
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1MiscAuthServerGet(options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1MiscAuthServerGet(options);
            return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
    };
};
exports.MiscApiFp = MiscApiFp;
/**
 * MiscApi - factory interface
 * @export
 */
const MiscApiFactory = function (configuration, basePath, axios) {
    const localVarFp = exports.MiscApiFp(configuration);
    return {
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1MiscAuthServerGet(options) {
            return localVarFp.apiV1MiscAuthServerGet(options).then((request) => request(axios, basePath));
        },
    };
};
exports.MiscApiFactory = MiscApiFactory;
/**
 * MiscApi - object-oriented interface
 * @export
 * @class MiscApi
 * @extends {BaseAPI}
 */
class MiscApi extends base_1.BaseAPI {
    /**
     *
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof MiscApi
     */
    apiV1MiscAuthServerGet(options) {
        return exports.MiscApiFp(this.configuration).apiV1MiscAuthServerGet(options).then((request) => request(this.axios, this.basePath));
    }
}
exports.MiscApi = MiscApi;
