// @dasha.ai/sdk/dist/rest-api/generated/core/api/sipendpoint-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Dasha.AI Platform API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SIPEndpointApi = exports.SIPEndpointApiFactory = exports.SIPEndpointApiFp = exports.SIPEndpointApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * SIPEndpointApi - axios parameter creator
 * @export
 */
const SIPEndpointApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @param {string} name
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointByNameGet: async (name, options = {}) => {
            // verify required parameter 'name' is not null or undefined
            common_1.assertParamExists('apiV1SipEndpointByNameGet', 'name', name);
            const localVarPath = `/api/v1/sip/endpoint/by-name`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            await common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            await common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            if (name !== undefined) {
                localVarQueryParameter['name'] = name;
            }
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointGet: async (skip, take, options = {}) => {
            const localVarPath = `/api/v1/sip/endpoint`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            await common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            await common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            if (skip !== undefined) {
                localVarQueryParameter['skip'] = skip;
            }
            if (take !== undefined) {
                localVarQueryParameter['take'] = take;
            }
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointIdDelete: async (id, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            common_1.assertParamExists('apiV1SipEndpointIdDelete', 'id', id);
            const localVarPath = `/api/v1/sip/endpoint/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'DELETE', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            await common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            await common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointIdGet: async (id, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            common_1.assertParamExists('apiV1SipEndpointIdGet', 'id', id);
            const localVarPath = `/api/v1/sip/endpoint/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            await common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            await common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {string} id
         * @param {SIPEndpointPatchDTO} [sIPEndpointPatchDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointIdPatch: async (id, sIPEndpointPatchDTO, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            common_1.assertParamExists('apiV1SipEndpointIdPatch', 'id', id);
            const localVarPath = `/api/v1/sip/endpoint/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'PATCH', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            await common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            await common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json-patch+json';
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = common_1.serializeDataIfNeeded(sIPEndpointPatchDTO, localVarRequestOptions, configuration);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @param {SIPEndpointAddDTO} [sIPEndpointAddDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointPost: async (sIPEndpointAddDTO, options = {}) => {
            const localVarPath = `/api/v1/sip/endpoint`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            await common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            await common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json-patch+json';
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = common_1.serializeDataIfNeeded(sIPEndpointAddDTO, localVarRequestOptions, configuration);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.SIPEndpointApiAxiosParamCreator = SIPEndpointApiAxiosParamCreator;
/**
 * SIPEndpointApi - functional programming interface
 * @export
 */
const SIPEndpointApiFp = function (configuration) {
    const localVarAxiosParamCreator = exports.SIPEndpointApiAxiosParamCreator(configuration);
    return {
        /**
         *
         * @param {string} name
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1SipEndpointByNameGet(name, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1SipEndpointByNameGet(name, options);
            return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         *
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1SipEndpointGet(skip, take, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1SipEndpointGet(skip, take, options);
            return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         *
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1SipEndpointIdDelete(id, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1SipEndpointIdDelete(id, options);
            return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         *
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1SipEndpointIdGet(id, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1SipEndpointIdGet(id, options);
            return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         *
         * @param {string} id
         * @param {SIPEndpointPatchDTO} [sIPEndpointPatchDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1SipEndpointIdPatch(id, sIPEndpointPatchDTO, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1SipEndpointIdPatch(id, sIPEndpointPatchDTO, options);
            return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         *
         * @param {SIPEndpointAddDTO} [sIPEndpointAddDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async apiV1SipEndpointPost(sIPEndpointAddDTO, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.apiV1SipEndpointPost(sIPEndpointAddDTO, options);
            return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
    };
};
exports.SIPEndpointApiFp = SIPEndpointApiFp;
/**
 * SIPEndpointApi - factory interface
 * @export
 */
const SIPEndpointApiFactory = function (configuration, basePath, axios) {
    const localVarFp = exports.SIPEndpointApiFp(configuration);
    return {
        /**
         *
         * @param {string} name
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointByNameGet(name, options) {
            return localVarFp.apiV1SipEndpointByNameGet(name, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {number} [skip]
         * @param {number} [take]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointGet(skip, take, options) {
            return localVarFp.apiV1SipEndpointGet(skip, take, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointIdDelete(id, options) {
            return localVarFp.apiV1SipEndpointIdDelete(id, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointIdGet(id, options) {
            return localVarFp.apiV1SipEndpointIdGet(id, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} id
         * @param {SIPEndpointPatchDTO} [sIPEndpointPatchDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointIdPatch(id, sIPEndpointPatchDTO, options) {
            return localVarFp.apiV1SipEndpointIdPatch(id, sIPEndpointPatchDTO, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {SIPEndpointAddDTO} [sIPEndpointAddDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1SipEndpointPost(sIPEndpointAddDTO, options) {
            return localVarFp.apiV1SipEndpointPost(sIPEndpointAddDTO, options).then((request) => request(axios, basePath));
        },
    };
};
exports.SIPEndpointApiFactory = SIPEndpointApiFactory;
/**
 * SIPEndpointApi - object-oriented interface
 * @export
 * @class SIPEndpointApi
 * @extends {BaseAPI}
 */
class SIPEndpointApi extends base_1.BaseAPI {
    /**
     *
     * @param {string} name
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof SIPEndpointApi
     */
    apiV1SipEndpointByNameGet(name, options) {
        return exports.SIPEndpointApiFp(this.configuration).apiV1SipEndpointByNameGet(name, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {number} [skip]
     * @param {number} [take]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof SIPEndpointApi
     */
    apiV1SipEndpointGet(skip, take, options) {
        return exports.SIPEndpointApiFp(this.configuration).apiV1SipEndpointGet(skip, take, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof SIPEndpointApi
     */
    apiV1SipEndpointIdDelete(id, options) {
        return exports.SIPEndpointApiFp(this.configuration).apiV1SipEndpointIdDelete(id, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof SIPEndpointApi
     */
    apiV1SipEndpointIdGet(id, options) {
        return exports.SIPEndpointApiFp(this.configuration).apiV1SipEndpointIdGet(id, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} id
     * @param {SIPEndpointPatchDTO} [sIPEndpointPatchDTO]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof SIPEndpointApi
     */
    apiV1SipEndpointIdPatch(id, sIPEndpointPatchDTO, options) {
        return exports.SIPEndpointApiFp(this.configuration).apiV1SipEndpointIdPatch(id, sIPEndpointPatchDTO, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {SIPEndpointAddDTO} [sIPEndpointAddDTO]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof SIPEndpointApi
     */
    apiV1SipEndpointPost(sIPEndpointAddDTO, options) {
        return exports.SIPEndpointApiFp(this.configuration).apiV1SipEndpointPost(sIPEndpointAddDTO, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.SIPEndpointApi = SIPEndpointApi;
